mesonbuild/scripts/msgfmthelper.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2016 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import subprocess
+
+parser = argparse.ArgumentParser()
+parser.add_argument('input')
+parser.add_argument('output')
+parser.add_argument('type')
+parser.add_argument('podir')
+parser.add_argument('--datadirs', default='')
+
+
+def run(args):
+    options = parser.parse_args(args)
+    env = {'GETTEXTDATADIRS': options.datadirs} if options.datadirs else None
+    return subprocess.call(['msgfmt', '--' + options.type, '-d', options.podir,
+                            '--template', options.input,  '-o', options.output],
+                           env=env)
```

Comment:
This unsets every other envvar (including PATH). Python docs say the following:

> these are used instead of the default behavior of inheriting the current process’ environment

What you probably meant was to add the setting in which case you would do `os.environ.copy()` and then `update()` it as needed.